packages/flame/lib/src/experimental/geometry/shapes/shape.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:ui';
+
+import 'package:vector_math/vector_math_64.dart';
+
+import '../../../game/transform2d.dart';
+import 'circle.dart';
+import 'polygon.dart';
+import 'rectangle.dart';
+import 'rounded_rectangle.dart';
+
+/// Base class for various 2D geometric primitives defined on a Cartesian
+/// coordinate plane.
+///
+/// Implementations include:
+///   - [Circle]
+///   - [Polygon]
+///   - [Rectangle]
+///   - [RoundedRectangle]
+abstract class Shape {
```

Comment:
Maybe we discussed this earlier but I forgot, shouldn't the shapes have an angle?

Reply:
The project() method can apply an arbitrary transform to an existing shape, including a rotation. The method also has the target= parameter to do the transformation in-place.